ext/dcos-installer/dcos_installer/installer_analytics.py

```diff
@@ -2,33 +2,36 @@
 import os
 import uuid
 
+from dcos_installer import backend
+
 
 class InstallerAnalytics():
-    def __init__(self, config):
+    def __init__(self):
         self.uuid = self.new_uuid()
-        self.bootstrap_version = os.getenv("BOOTSTRAP_ID")
-        self.customer_key = config["customer_key"] if 'customer_key' in config else ""
+        self.bootstrap_version = os.getenv("BOOTSTRAP_ID", "")
+        self.customer_key = ""
         self.source = "onprem"  # If you're using the installer you're not in AWS or Azure
 
-    def send(self, action, install_method, num_errors):
+    def send(self, action="", install_method="", num_errors=0):
         """Sends analytics track data to segmentIO.
         variant: string | open or enterprise
         action: string | preflight, deploy, or postflight
         install_method: string | gui, cli or advanced
         """
         analytics.write_key = "39uhSEOoRHMw6cMR6st9tYXDbAL3JSaP"
 
+        # We set customer key from config to avoid loading the config during class init
+        self.customer_key = backend.get_config()["customer_key"] if 'customer_key' in backend.get_config() else ""
+
         analytics.track(user_id=self.customer_key, anonymous_id=self.uuid, event=action, properties={
```

Comment:
I'm going to default it to None instead of "" since that's the module default: https://github.com/segmentio/analytics-python/blob/master/analytics/client.py#L64

This way we avoid two tracks in `if` statements.